plugins/language_java.lua

```diff
@@ -6,19 +6,21 @@ syntax.add {
   files = { "%.java$" },
   comment = "//",
   patterns = {
-    { pattern = "//.-\n",               type = "comment"  },
-    { pattern = { "/%*", "%*/" },       type = "comment"  },
-    { pattern = { '"', '"', '\\' },     type = "string"   },
-    { pattern = { "'", "'", '\\' },     type = "string"   },
-    { pattern = "'\\x%x?%x?%x?%x'",         type = "string"   }, -- character hexadecimal escape sequence
-    { pattern = "'\\u%x%x%x%x'",            type = "string"   }, -- character unicode escape sequence
-    { pattern = "'\\?.'",                   type = "string"   }, -- character literal
-    { pattern = "-?0x%x+",              type = "number"   },
-    { pattern = "-?%d+[%d%.eE]*f?",     type = "number"   },
-    { pattern = "-?%.?%d+f?",           type = "number"   },
-    { pattern = "[%+%-=/%*%^%%<>!~|&]", type = "operator" },
-    { pattern = "[%a_][%w_]*%f[(]",     type = "function" },
-    { pattern = "[%a_][%w_]*",          type = "symbol"   },
+    { pattern = "//.*",                                 type = "comment"  },
+    { pattern = { "/%*", "%*/" },                       type = "comment"  },
+    { pattern = { '"', '"', '\\' },                     type = "string"   },
+    { pattern = { "'", "'", '\\' },                     type = "string"   },
+    { pattern = "'\\x%x?%x?%x?%x'",                     type = "string"   }, -- character hexadecimal escape sequence
+    { pattern = "'\\u%x%x%x%x'",                        type = "string"   }, -- character unicode escape sequence
+    { pattern = "'\\?.'",                               type = "string"   }, -- character literal
+    { pattern = "-?0x%x+",                              type = "number"   },
+    { pattern = "-?%d+[%d%.eE]*f?",                     type = "number"   },
+    { pattern = "-?%.?%d+f?",                           type = "number"   },
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",                 type = "operator" },
+    { pattern = "[%a_][%w_]*%f[(]",                     type = "function" },
+    { regex   = "[A-Za-z]+(?=\\s[a-z0-9]+\\s\\=\\s)",   type = "function" }, -- Class name in class instance
```

Comment:
Why is this using a non-capturing group?

Reply:
Should it use `\s+` so that multiple spaces are still valid?

Reply:
Now that you mention it, yes it should.

Reply:
Should it also allow for underscores in the middle of the words?

Reply:
I've never seen class names with underscores.

Reply:
Yeah, but it's just a convention afaik.

Reply:
Mmh, ok I'll add it.